### PR TITLE
Add simple Docker build

### PR DESCRIPTION
**Pull request by ddgenome**
Add ability to build simple Docker images, i.e., those that have no
prerequisite steps to perform.  These goals can build and push
atomist/sdm-base.

### CODE REVIEW

File: lib/machine/dockerSupport.ts [hunk 1]
```diff
@@ -42,20 +52,73 @@ import { executeReleaseDocker } from "./release";
...
+            goalExecutor: executeReleaseDocker(sdm.configuration.sdm.docker.hub as DockerOptions),
+            pushTest: simpleDockerPushTest,
+            logInterpreter: LogSuppressor,
+        });
+
+    releaseVersion.with({
+        name: "file-release-version",
+        goalExecutor: executeReleaseVersion(fileProjectIdentifier,
+            // tslint:disable-next-line:no-invalid-template-strings
+            { command: "bash", args: ["-c", 'if [[ -f VERSION ]];then v=$(<VERSION);p=${v##*.};echo "${v%.*}.$((++p))" >VERSION;fi'] }),
```
Comment (cdupuis) [inline lib/machine/dockerSupport.ts]: What??? 🥇

File: lib/machine/dockerSupport.ts [hunk 2]
```diff
@@ -42,20 +52,73 @@ import { executeReleaseDocker } from "./release";
...
+    version.with({
+        name: "file-versioner",
+        versioner: async (sdmGoal, p, log) => {
+            const baseVersion: string = (await fileProjectIdentifier(p)).version;
+            log.write(`Using base version '${baseVersion}'`);
+            const branch = sdmGoal.branch;
+            const branchSuffix = (branch === sdmGoal.push.repo.defaultBranch) ? "" :
+                "branch-" + branch.replace(/[_/]/g, "-") + ".";
+            log.write(`Commit is on branch '${branch}', using '${branchSuffix}'`);
+            const ts = dateformat(new Date(), "yyyymmddHHMMss", true);
```
Comment (cdupuis) [inline lib/machine/dockerSupport.ts]: There's a `formatDate` helper function in sdm that could use instead here.

File: lib/machine/dockerSupport.ts [hunk 3]
```diff
@@ -42,20 +52,73 @@ import { executeReleaseDocker } from "./release";
...
+            { command: "bash", args: ["-c", 'if [[ -f VERSION ]];then v=$(<VERSION);p=${v##*.};echo "${v%.*}.$((++p))" >VERSION;fi'] }),
         logInterpreter: LogSuppressor,
+        pushTest: simpleDockerPushTest,
     });
 
     return sdm;
 }
+
+const fileProjectIdentifier: ProjectIdentifier = async p => {
+    const versionFile = await p.getFile("VERSION");
```
Comment (cdupuis) [inline lib/machine/dockerSupport.ts]: In the manifesto-sdm I used `.version` as file with very similar code.

File: lib/machine/dockerSupport.ts (hunk 1 above)
Comment (ddgenome) [reply lib/machine/dockerSupport.ts]: This was the easiest way, `executeReleaseVersion` wants a spawn command.

File: lib/machine/dockerSupport.ts (hunk 3 above)
Comment (ddgenome) [reply lib/machine/dockerSupport.ts]: Would you prefer I use `.version` instead of `VERSION`?

File: lib/machine/dockerSupport.ts (hunk 2 above)
Comment (ddgenome) [reply lib/machine/dockerSupport.ts]: I'll take a look.

Comment (ddgenome) [reply lib/machine/dockerSupport.ts]: That function uses the system time zone, not UTC.

https://github.com/atomist/sdm/blob/03c448917d1bbc4bb880dca74cb5b495ef9fe420/lib/api-helper/misc/dateFormat.ts#L27

Do you think I should submit a PR in sdm to change that?

**Comment by lievendoclo** [comment]
Shouldn't this go in the docker pack?

File: lib/machine/dockerSupport.ts (hunk 2 above)
Comment (cdupuis) [reply lib/machine/dockerSupport.ts]: Yeah, that's probably a good idea. Change in master is fine though.

File: lib/machine/dockerSupport.ts (hunk 3 above)
Comment (cdupuis) [reply lib/machine/dockerSupport.ts]: Now that I see both I prefer VERSION. Sorry for the noise.

File: lib/machine/dockerSupport.ts (hunk 2 above)
Comment (ddgenome) [reply lib/machine/dockerSupport.ts]: Too late

**Comment by ddgenome** [comment]
Which parts? The stuff I added is analogous to stuff already in atomist-sdm.

**Comment by ddgenome** [comment]
I've typically gotten stuff working in atomist-sdm and then migrated to packs as appropriate.

**Comment by ddgenome** [comment]
The only real code added here is the file-based version stuff, which does not belong in sdm-pack-docker.

**Pull request merged**